### ISSUE TITLE
Duplicate results

### ISSUE THREAD

**Issue by pepa65** [useful-forks/useful-forks.github.io#59]
Not sure if this is to be expected somehow, but I noticed nothing in the Issues about this: I'm getting lots of duplicate results displayed, the same lines with the same values. When exported to csv, it is the same.

For example, the file `useful-forks-https\ __github.com_zyedidia_micro_.csv` has 104 lines (including the header), while with duplicate lines removed, it only has 59.

**Comment by payne911** [comment]
Good catch. For posterity, here's the result that this user is talking about: https://justpaste.it/8evui
I do see a bunch of duplicates. I'll look into it.

**Comment by Ethkuil** [comment]
> The simplest fix to this is to keep a HashSet of all entries added in the table, and to only append new rows if the repo isn't in the HashSet.

The even better fix is to change [`TABLE_DATA`](https://github.com/useful-forks/useful-forks.github.io/pull/58/commits/5428fa031d22ebcc62e31ff180a15aa2d56df55f#diff-99957638fc26865826a396d074c87a7a2eb51945baf96f6505c805d3386ea1c3R5) into a set instead of an array.

**Comment by payne911** [comment]
> to change [TABLE_DATA](https://github.com/useful-forks/useful-forks.github.io/pull/58/commits/5428fa031d22ebcc62e31ff180a15aa2d56df55f#diff-99957638fc26865826a396d074c87a7a2eb51945baf96f6505c805d3386ea1c3R5) into a set instead of an array.

Theoretically, a set has no ordering. Thus, the operation of sorting would make no sense against that data structure.

Also, it seems like JavaScript has no built-in set (and thus I regret saying "The simplest fix to this is to keep a HashSet"). We'd have to make one, and at that point it might become not worth it: the extra code increases complexity and reduces maintainability, while not providing much performance gains because we are talking about relatively small tables.

The simplest approach would be to simply scan the entire table array to check for a duplicate: that's an acceptable `O(n)`.

**Comment by payne911** [comment]
Update: I have the fix ready. I'm simply waiting on #58 to get merged in as it'll build on it.

**Comment by payne911** [comment]
Odd. This issue is still happening, but it's not consistent.
![image](https://user-images.githubusercontent.com/38117856/227691666-258103cb-3fad-4110-b600-acbc6e3b1812.png)

There's most probably a race-condition happening. Could be in the Octokit library, or straight in my code. I'll have to try to manually bisect.

Or maybe my JavaScript knowledge is too narrow for me to even find the source of this bug.  ¯\_(ツ)_/¯

**Comment by payne911** [comment]
https://github.com/useful-forks/useful-forks.github.io/commit/229d0ce308dc47795d7323af9b0a128a0ef7f4ec

This was what I thought would fix it.

**Comment by andy0130tw** [comment]
It's this the set you want?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

**Comment by payne911** [comment]
Yes, although technically given [I have done an implementation with a list](https://github.com/useful-forks/useful-forks.github.io/commit/229d0ce308dc47795d7323af9b0a128a0ef7f4ec) which should have solved this issue, it is clear that the problem is not what I thought it was.

> There's most probably a race-condition happening. Could be in the Octokit library, or straight in my code

Feel free to investigate as I do not have much free time these days.
